src/poetry/console/commands/install.py

```diff
@@ -79,12 +80,23 @@ def handle(self) -> int:
             self.poetry.config.get("experimental.new-installer", False)
         )
 
-        extras = []
-        for extra in self.option("extras"):
-            if " " in extra:
-                extras += [e.strip() for e in extra.split(" ")]
-            else:
-                extras.append(extra)
+        if self.option("extras") and self.option("all-extras"):
+            self.line_error(
+                "<error>You cannot specify explicit"
+                " `<fg=yellow;options=bold>--extras</>` while installing"
+                " `<fg=yellow;options=bold>--all-extras</>`.</error>"
+            )
+            return 1
```

Comment:
What do you think of `--all-extras` instead overriding `--extras`, maybe emitting a warning rather than an error when both are specified?

Reply:
I'd rather throw the error to prevent unexpected behavior for inattentive users.